### PR TITLE
feat: Add pre-commit hook for lint

### PR DESCRIPTION
**Pull request by kushthedude**

### CODE REVIEW

File: ui/package.json [hunk 1]
```diff
@@ -13,6 +13,10 @@
     "cy:run": "cypress run",
     "test": "start-server-and-test dev http://localhost:3000 cy:run && eslint ."
   },
+  "pre-commit": [
+    "lint",
```
Comment (leecalcote) [inline ui/package.json:17]: Does this pre-commit hook require that contributors separately install `eslint` or will `eslint --fix` be included with the rest of the NPM packages as part of building this repo?

Comment (leecalcote) [reply ui/package.json:17]: Do we have an update to the Contributor guide to accompany this?

Comment (kushthedude) [reply ui/package.json:17]: > Does this pre-commit hook require that contributors separately install `eslint` or will `eslint --fix` be included with the rest of the NPM packages as part of building this repo?

While setting up the repository on local, we have `make setup-ui-libs` which by default installs the DevDependencies too. So we do not have to worry about the inclusion of ES-Lint in local setup, we have only ignored DevDependencies in our Production Environment to make our image size less.

Comment (leecalcote) [reply ui/package.json:17]: 👍

**Comment by Nikhil-Ladha** [comment]
Is pre-push needed, as pre-commit is there already?

**Comment by kushthedude** [comment]
> Is pre-push needed, as pre-commit is there already?

Not every push is preceded by commit, for example when you merge your local fork with upstream when branches are inter-merged when a release tag is made from cli, when a new branch is made from cli and pushed and more other events, so for these actions, a pre-push is defined/

**Review by Nikhil-Ladha — APPROVED** [review]
Ohh. Gotcha!
Look's good 👍🏼

**Comment by leecalcote** [comment]
@kushthedude will you present this at the Dev meeting on Wednesday as small bit of training for all on what to expect and what to do when a pre-comment issue rears up?

**Comment by kushthedude** [comment]
Sure thing Sir 🚀

**Pull request merged**